### PR TITLE
Add adaptive auth docs

### PR DESCRIPTION
**Pull request by DMHP**
## Purpose
Add adaptive auth docs

### CODE REVIEW

File: en/mkdocs.yml [hunk 1]
```diff
@@ -191,21 +193,21 @@ nav:
           - 'JDBC as the Primary Userstore': guides/userstore/jdbc-us.md
           - 'Active Directory as a Secondary Userstore': guides/userstore/ad-us.md
           - 'LDAP as a Secondary Userstore': guides/userstore/ldap-us.md
-          - 'Plug a Custom Userstore': guides/userstore/custom-us.md
+          - 'Plug a Custom User store': guides/userstore/custom-us.md
```
Comment (sherenewso2) [inline en/mkdocs.yml:196]: store --> Store

File: en/docs/guides/adaptive-auth/configure-adaptive-auth.md [hunk 2]
```diff
@@ -78,4 +69,9 @@ function someCommonFunction(context) {
 
 - To learn about the guidelines on writing custom functions for adaptive authentication, see [Writing Custom Functions for Adaptive Authentication](insertlink).
 
-----
\ No newline at end of file
+
+!!! info "Related Topics"
+    - [Concept: Adaptive-Authentication](TODO:insert-link-to-concept)
+    - [Guide: Ensure Assurance with Authentication Context Class Reference (ACR)  and Authentication Method Reference (AMR)](../../adaptive-auth/work-with-acr-amr)
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/configure-adaptive-auth.md:75]: can we shorten this to "Ensure Assurance with ACR/AMR" like we have titled it in the mkdocs.yml ?

File: en/docs/fragments/register-an-application-request-path.md [hunk 3]
```diff
@@ -2,30 +2,8 @@
...
-11.	Select all the options given under **Allowed grant type**.
-
-12. Navigate to the **Sign-on Method** tab.
+{!fragments/register-playground-application-portal.md!}
 
 13. Check **basic-auth** under **Request Path Authentication**.
 
-14.	Click **Update**.
\ No newline at end of file
+{!fragments/save-sp-in-portal.md!}
```
Comment (sherenewso2) [inline en/docs/fragments/register-an-application-request-path.md:9]: This need not be a fragment...we can just leave it as step 14.

File: en/docs/fragments/register-an-application-oauth-request-path.md [hunk 4]
```diff
@@ -2,30 +2,8 @@
...
-11.	Select all the options given under **Allowed grant type**.
-
-12. Navigate to the **Sign-on Method** tab.
+{!fragments/register-playground-application-portal.md!}
 
 13. Check **oauth-bearer** under **Request Path Authentication**.
 
-14.	Click **Update**.
\ No newline at end of file
+{!fragments/save-sp-in-portal.md!}
```
Comment (sherenewso2) [inline en/docs/fragments/register-an-application-oauth-request-path.md:9]: This need not be a fragment...we can just leave it as step 14.

File: en/docs/fragments/save-sp-in-portal.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+After all the configurations click **Update**.
```
Comment (sherenewso2) [inline en/docs/fragments/save-sp-in-portal.md:1]: This need not be a fragment...we can just have it as a step on the relevant page itself right?

File: en/docs/guides/adaptive-auth/adaptive-auth-with-function-lib.md [hunk 6]
```diff
@@ -197,4 +197,10 @@ Follow the instructions below to delete a function library.
     ```tab="Example"
     ageModule.getAge(birthday)
     ```
+    
+!!! info "Related Topics"
+    - [Concept: Adaptive-Authentication](TODO:insert-link-to-concept)
+    - [Guide: Configure Adaptive Authentication for an Application](../../guides/adaptive-auth/configure-adaptive-auth)
+    - [Guide: Ensure Assurance with Authentication Context Class Reference (ACR)  and Authentication Method Reference (AMR)](../../guides/adaptive-auth/work-with-acr-amr)
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/adaptive-auth-with-function-lib.md:204]: can we shorten this to "Ensure Assurance with ACR/AMR" like we have titled it in the mkdocs.yml ?

File: en/docs/guides/adaptive-auth/configure-adaptive-auth.md [hunk 7]
```diff
@@ -6,36 +6,27 @@ This page guides you through setting up [adaptive authentication](insertlink) fo
 
 This guide assumes you have your own application. If you wish to try out this flow with a sample application, click the button below. 
 
-<a class="samplebtn_a" href="../../../quick-starts/role-based-adaptive-auth"   rel="nofollow noopener">Try it with the sample</a>
+<a class="samplebtn_a" href="../../../quick-starts/adaptive-auth-overview"   rel="nofollow noopener">Try it with the sample</a>
+
+!!! note
+    Following scenarios are available as samples.
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/configure-adaptive-auth.md:12]: The following scenarios..

File: en/docs/guides/adaptive-auth/configure-adaptive-auth.md [hunk 8]
```diff
@@ -6,36 +6,27 @@ This page guides you through setting up [adaptive authentication](insertlink) fo
...
-    ![authentication templates](../assets/img/guides/authentication-templates.png)
-
-    !!! tip
-        To see a sample flow of each template, see [Adaptive Authentication Scenarios](../../../quick-starts/adaptive-auth-samples).
-
-6.  Click **Ok**. 
-    The authentication script and authentication steps are configured. 
-
-----
+{!fragments/add-adaptive-script-portal.md!}
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/configure-adaptive-auth.md:29]: This fragment only contains steps specific to ACR-based template. This fragment doesn't fit here since this page is the generic guide for all adaptive auth scenarios.

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
+# Ensure Assurance with Authentication Context Class Reference (ACR)  and Authentication Method Reference (AMR)
+
+This page guides you through enforcing additional assurance in the user authentication flow using **Authentication Context Class Reference** (ACR)  and **Authentication Method Reference** (AMR)
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:3]: shall we add a placeholder here to link to a concept doc

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+This guide assumes you have your own application. If you wish to try out this flow with a sample application, click the button below. 
+
+<a class="samplebtn_a" href="../../../quick-starts/acr-based-adaptive-auth" rel="nofollow noopener">Try it with the sample</a>
+
+----
+
+{!fragments/register-playground-application-portal.md!}
+
+{!fragments/add-adaptive-script-portal.md!}
+Replace the authentication script added from the template with the following.
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:16]: "with the following"
is there a missing code block that should be added here?

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+		  if(needLevel2) {
+		    executeStep(3);
+		    context.selectedAcr="LOA2";
+		  }
+		  if(needLevel3) {
+		    executeStep(4);
+		    context.selectedAcr="LOA3";
+		  }
+		}
+
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:47]: add separator

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+		  if(needLevel3) {
+		    executeStep(4);
+		    context.selectedAcr="LOA3";
+		  }
+		}
+		
+## Configure the client application
+Send the following requests via your application to connect your application to WSO2 IS.
+
+1. To initiate the authentication flow, update the placeholders and send following request to the authorization endpoint.
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:51]: send following request --> send the following request

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
...ken":"60a66d57-0e48-3896-98e7-00213acee104","scope":"openid","id_token":"eyJ4NXQiOiJNell4TW1Ga09HWXdNV0kwWldObU5EY3hOR1l3WW1NNFpUQTNNV0kyTkRBelpHUXpOR00wWkdSbE5qSmtPREZrWkRSaU9URmtNV0ZoTXpVMlpHVmxOZyIsImtpZCI6Ik16WXhNbUZrT0dZd01XSTBaV05tTkRjeE5HWXdZbU00WlRBM01XSTJOREF6WkdRek5HTTBaR1JsTmpKa09ERmtaRFJpT1RGa01XRmhNelUyWkdWbE5nX1JTMjU2IiwiYWxnIjoiUlMyNTYifQ.eyJhdF9oYXNoIjoiVFdlYWp5dE1TQU0zX1k3Q09mQm4yUSIsImF1ZCI6IjV2T1FaVmZvT0hMbXRYYk5ISnBZWGM3Wkx1Y2EiLCJhY3IiOiJhY3IyIiwiY19oYXNoIjoiVFN1b0Z2eHlybEdCZkxQZW1ZbEt3USIsInN1YiI6ImFkbWluIiwibmJmIjoxNTkxOTQ0ODk1LCJhenAiOiI1dk9RWlZmb09ITG10WGJOSEpwWVhjN1pMdWNhIiwiYW1yIjpbIkJhc2ljQXV0aGVudGljYXRvciJdLCJpc3MiOiJodHRwczpcL1wvbG9jYWxob3N0Ojk0NDNcL29hdXRoMlwvdG9rZW4iLCJleHAiOjE1OTE5NDg0OTUsImlhdCI6MTU5MTk0NDg5NSwic2lkIjoiMjBmODlmYzctNDRkZC00MjNkLTlkNDktZGRjMTVlNWVmNGRlIn0.d7WpY220UTZ10zCg-DBmKr-0f0k2tbYk00xqJoTmN2oNmP5u6x8_kUIASSeeP2p1LDeHuT6IQb045YCJDStsiSpwD0XUtTrFPJoRFUiIDgzXBQCBz8pzCjv2w-AMj8hJu965nSbwqCt-20OjKDPb187jiMyAxYMpU0o9Zk470whrkZQkC2SA16KhKdrIwpHHiiOI0pX-LdSsrZFzsw2ZWQmlzdRh4xoN6wTsLI9jhxz52mqs0Ghea9MlVFeVuIf6BeFN8ZpwzsnpVbSO9g4ZCZVYz3dtuiIaBQgoYy3E0SMG1QdxGgg7nYg0NQd-wfInYuhik0BSSrLKmJhff5YHaQ","token_type":"Bearer","expires_in":2554}
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:97]: These code blocks might need to be indented out so that it doesn't come inside the info macro...please check

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+!!! note
+    Following is a sample SAML fragment that you can add to the SAML SSO request while initiating the SAML SSO flow. 
+
+    ```
+    <samlp:RequestedAuthnContext Comparison="exact">
+    <saml:AuthnContextClassRef>urn:federation:authentication:windows</saml:AuthnContextClassRef>
+    <saml:AuthnContextClassRef>pwd</saml:AuthnContextClassRef>
+    <saml:AuthnContextClassRef>LOA2</saml:AuthnContextClassRef>
+    </samlp:RequestedAuthnContext>
+    ```
```
Comment (sherenewso2) [inline en/docs/guides/adaptive-auth/work-with-acr-amr.md:151]: can this be an optional step instead of a note?

File: en/docs/quick-starts/adaptive-auth-overview.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+   - [ACR-Based](acr-based-adaptive-auth.md)
+   - [Risk-Based](risk-based-adaptive-auth.md)
+   - [With Function Library](adaptive-auth-with-function-lib-sample.md)
+   - [Limiting Active User Sessions](limiting-active-sessions-adaptive-auth.md)
+   
+!!! info "Related Topics"
+    - [Concept: Adaptive-Authentication](TODO:insert-link-to-concept)
+    - [Guide: Configure Adaptive Authentication for an Application](../../guides/adaptive-auth/configure-adaptive-auth)
+    - [Guide: Adaptive Authentication Using Function Library](../../guides/adaptive-auth/adaptive-auth-with-function-lib)
+    - [Guide: Ensure Assurance with Authentication Context Class Reference (ACR)  and Authentication Method Reference (AMR)](../../guides/adaptive-auth/work-with-acr-amr)
```
Comment (sherenewso2) [inline en/docs/quick-starts/adaptive-auth-overview.md:20]: can we shorten this to "Ensure Assurance with ACR/AMR" like we've titled it in the mkdocs yaml?

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md (hunk 14 above)
Comment (DMHP) [reply en/docs/guides/adaptive-auth/work-with-acr-amr.md:151]: This is not an optional step. This is a note on how to send a request with SAML.

File: en/docs/guides/adaptive-auth/work-with-acr-amr.md (hunk 13 above)
Comment (DMHP) [reply en/docs/guides/adaptive-auth/work-with-acr-amr.md]: In the PR it seems like the code is not indented. But actually it does in the IDE and it displays properly in the site.
